cocos/gfx/device-manager.ts

```diff
@@ -100,16 +108,23 @@ export class DeviceManager {
         return this._swapchain;
     }
 
-    public init (canvas: HTMLCanvasElement | null, bindingMappingInfo: BindingMappingInfo): void {
+    private async _tryInitializeDevice (DeviceConstructor, info: DeviceInfo): Promise<void> {
+        if (!this._initDevice && DeviceConstructor) {
+            this._gfxDevice = new DeviceConstructor();
+            this._initDevice = await this._gfxDevice.initialize(info);
+        }
+    }
+
+    public async init (canvas: HTMLCanvasElement | null, bindingMappingInfo: BindingMappingInfo): Promise<void> {
```

Comment:
Consider keeping `init` as a normal function.
We might add a new function like `public async requestAdapter (xxx): Promise<void>` to call WebGPU async functions.
The reason is that `async` is contagious and will change code execution flow, which might break compatibility.